inmoose/limma/lmfit.py

```diff
@@ -0,0 +1,435 @@
+        fitting method: :code:`"ls"` for least squares or :code:`"robust"` for
+        robust regression
+
+    Returns
+    -------
+    MArrayLM
+        object containing the result of the fits.
+        The row names of :code:`obj` are preserved in the fit object and can be
+        retrieved by :code:`fit.index` where :code:`fit` is the output of
+        :fun:`lmFit`.
+        The column names of :code:`design` are preserved as column names and
+        can be retrieved by :code:`fit.columns`.
+    """
+    # Fit genewise linear models
+
+    # Extract components from obj
+    y = EAWP(obj)
+
+    if y.exprs.shape[0] == 0:
+        raise ValueError("expression matrix has zero rows")
+
+    # Check design matrix
+    if design is None:
+        design = y.design
+    if design is None:
+        design = patsy.dmatrix("~1", data=y.exprs)
+    else:
+        if not isinstance(design, patsy.DesignMatrix):
+            raise ValueError("design must be a patsy DesignMatrix")
+        if design.shape[0] != y.exprs.shape[1]:
+            raise ValueError(
+                "row dimension of design does not match column dimension of data object"
+            )
+        if np.isnan(design).any():
+            raise ValueError("NAs not allowed in design matrix")
+
+    ne = nonEstimable(design)
+    if ne is not None:
+        logging.info(f"Coefficients not estimable: {' '.join(ne)}")
+
+    # Check ndups and spacing. Default to 1.
+    if ndups is None:
+        try:
+            ndups = y.printer.ndups
+        except AttributeError:
+            pass
+    if ndups is None:
+        ndups = 1
+    if spacing is None:
+        try:
+            spacing = y.printer.spacing
+        except AttributeError:
+            pass
+    if spacing is None:
+        spacing = 1
+
+    # Check weights
+    if weights is None:
+        weights = y.weights
+
+    # Check method
+    if method not in ["ls", "robust"]:
+        raise ValueError(f"unknown method: {method}")
+
+    # If duplicates are present, reduce probe-annotation and Amean to correct length
+    if ndups > 1:
+        if y.probes is not None:
+            y.probes = uniquegenelist(y.probes, ndups=ndups, spacing=spacing)
+        if y.Amean is not None:
+            y.Amean = np.nanmean(
+                unwrapdups(y.Amean, ndups=ndups, spacing=spacing), axis=1
+            )
+
+    # Dispatch fitting algorithms
+    if method == "robust":
+        raise NotImplementedError("robust method for lmFit is not implemented yet")
+        # fit = mrlm(
+        #    y.expres, design=design, ndups=ndups, spacing=spacing, weights=weights
+        # )
+    else:
+        if ndups < 2 and block is None:
+            fit = lm_series(
+                y.exprs, design=design, ndups=ndups, spacing=spacing, weights=weights
+            )
+        else:
+            if correlation is None:
+                raise ValueError(
+                    "the correlation must be set, see function duplicateCorrelation"
```

Comment:
Just to be sure, `duplicateCorrelation` function is not yet implemented and will may be in a future PR? Or it come from another module? (I couldn't find it in the code) 
Maybe we can put an `NotImplementedError`

Reply:
This `ValueError` informs the user that the `correlation` parameter is mandatory for duplicated channels.
However, the duplicated channels features are not implemented in `inmoose` for the time being.
The following line of code does raise a `NotImplementedError` to signal this.

Reply:
very clear thanks !